ast.ts

```diff
@@ -28,6 +28,17 @@ export type Stmt<A> =
   | {  a?: A, tag: "if", cond: Expr<A>, thn: Array<Stmt<A>>, els: Array<Stmt<A>> }
   | {  a?: A, tag: "while", cond: Expr<A>, body: Array<Stmt<A>> }
 
+// isSimple should be true when destruct has no comma(,)
+// e.g. a, = 1, -> isSimple = false
+// e.g. a = 1   -> isSimple = true
```

Comment:
I have an idea, do you think its possible to trim down things like this in parsing and make everything simplified so you won't need `isSimple` at all? I'm a bit worried that's gonna create an extra unnecessary flag for you which might not be really beneficial... For example in the example in your comment, imagine `a,` getting automatically simplified to `a` in parsing and also `1,`  to `1`.

Reply:
After careful consideration, I found it perhaps not suitable to remove `isSimple` field. See the below snippet:
```Python
>>> a, = [1]
>>> a
1
>>> a = [1]
>>> a
[1]
```
Without `isSimple` field, we could not distinguish the two scenarios.